### PR TITLE
move plugins out of experimental

### LINKED ISSUE #26760: Move Pluginv2 out of experimental
Pluginv2 is currently available on experimental daemon builds only. Once the intended feature set is complete and correct, move it out of experimental into stable. This is slated for 1.13.

### PR DESCRIPTION
**Pull request by vieux**
ping @anusha-ragunathan @tiborvass @thaJeztah 

closes #26760

### CODE REVIEW

**Comment by thaJeztah** [comment]
Can you check integration tests as well? Tests are probably skipped if experimental is not enabled

**Comment by vieux** [comment]
@thaJeztah right, thanks

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by stevvooe** [comment]
This cannot be merged until #28148 is merged.

**Comment by anusha-ragunathan** [comment]
Please add 26760, which tracks this.

**Comment by anusha-ragunathan** [comment]
tests that need to be updated:
- docker_cli_daemon_experimental_test.go (all of them)
- docker_cli_events_test.go (TestEventsPluginOps)
- docker_cli_authz_plugin_v2_test.go (all of them)
- docker_cli_network_unix_test.go (TestDockerPluginV2NetworkDriver)

**Comment by anusha-ragunathan** [comment]
`daemon/daemon_experimental.go` needs to remove the `daemon.HasExperimental` checks and merge with `daemon/daemon.go`. Else nothing will work.

File: docs/reference/api/docker_remote_api.md
Comment (anusha-ragunathan) [inline docs/reference/api/docker_remote_api.md]: Add plugin create:
``POST /plugins/create?name=plugin name` creates a plugin.`

Comment (anusha-ragunathan) [inline docs/reference/api/docker_remote_api.md]: Also missing `push`

File: docs/extend/config.md
Comment (anusha-ragunathan) [inline docs/extend/config.md]: Why is this being removed?

Comment (vieux) [inline docs/extend/config.md]: It's a doc left-over from a previous PR, it's already removed from the code, after talking with @tiborvass and @stevvooe. 

* You needed to json-decode in order to get the version, if the format change a lot, it won't even work.
* The version is already present in the media-type, that's how other media handle versioning.

**Comment by vieux** [comment]
Thanks @anusha-ragunathan, I missed a lot indeed

**Comment by anusha-ragunathan** [comment]
Thank you for doing this @vieux . I know you've been burning the midnight oil with a lotta plugin PRs!

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

File: docs/reference/api/docker_remote_api.md
Comment (anusha-ragunathan) [inline docs/reference/api/docker_remote_api.md]: typo `pluging`

File: plugin/distribution/pull.go
Comment (anusha-ragunathan) [inline plugin/distribution/pull.go]: A similar check for `image.v0` needs to be added in https://github.com/vieux/docker/blob/exit_exp_plugin/distribution/pull_v2.go#L361, to avoid using `docker pull` to pull plugin images.

Comment (vieux) [inline plugin/distribution/pull.go]: thanks

**Comment by anusha-ragunathan** [comment]
LGTM (once tests are all green)

**Comment by tiborvass** [comment]
@vieux needs rebase :(

**Comment by tiborvass** [comment]
apart from small conflict, LGTM

**Comment by jhowardmsft** [comment]
As plugins don't work on Windows, can we get this updated to report a nice "not supported on this platform" message for the API endpoints? And document accordingly.

**Comment by vieux** [comment]
@jhowardmsft yes it's on my list of things to do during the freeze

**Comment by jhowardmsft** [comment]
Thanks, @vieux

**Comment by tomdee** [comment]
@vieux Am I right in thinking that network plugins _still_ aren't supported?

**Comment by mavenugo** [comment]
@tomdee no. network plugins are supported. PTAL : https://github.com/docker/docker/blob/master/integration-cli/docker_cli_network_unix_test.go#L771

**Pull request merged**
